client/src/lib/firebase/splash.ts

```diff
@@ -0,0 +1,27 @@
+import { db } from "./app";
+import type { UserData } from "./firestore-types/users";
+import { doc, updateDoc, getDoc, setDoc } from "firebase/firestore";
+
+// users collection
+//  - displayName
+//  - avatar
+// get display name function to autofill the name input text
+export const getDisplayName = async (id: string) => {
+  const user = await getDoc(doc(db, `/users/${id}`));
+  return user.data() as UserData;
+};
+// allow user to changing after every keystroke
+export const saveDisplayName = async (id: string, name: string) => {
+  const user = doc(db, `/users/${id}`);
+  await updateDoc(user, {
+    displayName: name,
+  });
+};
+
+// create user document for anon user
+export const createAnonUser = async (uid: string, name: string) => {
```

Comment:
why is this called createAnonUser? isn't is setting a doc for any user?

Reply:
I assumed that if they signed in they should already have a user document in the users collection but your suggestion is good since it wasn't stated